docs/entities-intro.rst

```diff
@@ -124,11 +124,21 @@ Can I have millions of Entities?
 
 There are two current limitations that make millions of Entities impractical: data transfer and form performance.
 
-Currently, all Entities that have not been deleted are sent to every device on every update. Depending on your data connection, this may be a limiting factor for your project. We will eventually add support for archiving Entities to address this limitation.
+Currently, all Entities that have not been deleted are sent to every device on every update. Depending on your data connection, this may be a limiting factor for your project. To address this, we will eventually add functionality such as archiving Entities, synchronizing only updated Entities, or assigning a subset of Entities to a user.
 
-Entities are currently represented in memory for access by forms. Modern devices can easily process multiple tens of thousands of entities in this way, but your form may become slow or crash if you have more than 50,000 Entities.
+Starting in Collect v2024.3 and Central v2024.3, Entities are represented in a database. If you use simple expressions using ``=``, ``and`` and ``or`` in choice filters or :ref:`lookup expressions <referencing-values-in-datasets>`, they will be performed quickly. We verify that a form with 100,000 Entities with 6 properties that are each 36 random characters long is usable on a device from 2021 with 3 GB of RAM.
```

Comment:
Do we need to say "database"? It feels like too much detail. What about:
 
> Starting in Collect v2024.3 and Central v2024.3, Entities are are optimized so that larger numbers can be used in forms. There are limitations to this however.